### ISSUE TITLE
[BUG] v1.36.3 breaks PGFPlotsX inset position

### ISSUE THREAD

**Issue by ivan-boikov** [JuliaPlots/Plots.jl#4586]
## Details
The inset positioning in `PGFPlotsX` breaks between `v.1.36.2` and `v.1.36.3`. The bug persists until `v.1.37.1`, including. No change in `GR` and `PlotlyJS` and they are working fine, no change in `UnicodePlots` and `InspectDR`, but they did not work before.

```julia
using Plots, Pkg

function test()
    # https://discourse.julialang.org/t/how-to-find-out-the-version-of-a-package-from-its-module/37755
    pkgversion(m::Module) = Pkg.TOML.parsefile(joinpath(dirname(string(first(methods(m.eval)).file)), "..", "Project.toml"))["version"]
    plot(; title="Plots.jl v." * pkgversion(Plots));
    histogram!(rand(500), inset = (1, bbox(0.0,0.0,0.5,0.5)), subplot = 2,
        color=:red, alpha=0.5, framestyle=:none, background_color_subplot=:transparent);
    histogram!(randn(500), inset = (1, bbox(0.5,0.5,0.5,0.5)), subplot = 3,
        color=:green, alpha=0.5, framestyle=:none, background_color_subplot=:transparent)
    scatter!(randn(500), inset = (1, bbox(0.0,0.5,0.5,0.5)), subplot = 4,
        color=:violet, alpha=0.5, framestyle=:none, background_color_subplot=:transparent)
    plot!(randn(100), inset = (1, bbox(0.5,0.0,0.5,0.5)), subplot = 5,
        color=:black, alpha=0.5, framestyle=:none, background_color_subplot=:transparent)
end

# works on <=v.1.36.2, does not on >=v.1.36.3
pgfplotsx()
PGFPlotsX.latexengine!(PGFPlotsX.PDFLATEX) # doubt it's relevant, but just in case - I use it
test()
```
![pic-selected-221207-2042-17](https://user-images.githubusercontent.com/66747290/206281530-0c5b6a03-7c9f-46d5-8c1c-e5905ed68053.png)

![pic-selected-221207-2044-07](https://user-images.githubusercontent.com/66747290/206281594-cc6a63ad-7839-44dd-9273-af024cd0f72f.png)

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |     |  x   |
pyplot       |     |     | x
plotlyjs     |     |  x   |
pgfplotsx    |  x   |     |
unicodeplots |     |  x   |
inspectdr    |     |  x   |
gaston       |     |     | x

### Versions

Plots.jl version: 1.36.3
Backend  version (`]st -m <backend(s)>`):
On the working version (v1.36.2)
```
[28b8d3ca] GR v0.69.5
[4b11ee91] Gaston v1.0.5
[d0351b0e] InspectDR v0.4.3
[8314cec4] PGFPlotsX v1.5.1
[58dd65bb] Plotly v0.4.1
[f0f68f2c] PlotlyJS v0.18.10
[91a5bcdd] Plots v1.36.2
[d330b81b] PyPlot v2.11.0
[b8865327] UnicodePlots v2.12.4
```
On the broken version (v1.36.3)
```
[28b8d3ca] GR v0.69.5
[4b11ee91] Gaston v1.0.5
[d0351b0e] InspectDR v0.4.3
[8314cec4] PGFPlotsX v1.5.1
[58dd65bb] Plotly v0.4.1
[f0f68f2c] PlotlyJS v0.18.10
[91a5bcdd] Plots v1.36.3
[d330b81b] PyPlot v2.11.0
[b8865327] UnicodePlots v2.12.4
```
Output of `versioninfo()`:
```
Julia Version 1.8.1
Commit afb6c60d69a (2022-09-06 15:09 UTC)
Platform Info:
OS: Linux (x86_64-linux-gnu)
CPU: 12 × AMD Ryzen 5 3600 6-Core Processor
WORD_SIZE: 64
LIBM: libopenlibm
LLVM: libLLVM-13.0.1 (ORCJIT, znver2)
Threads: 12 on 12 virtual cores
Environment:
JULIA_DEPOT_PATH = /home/ivan/.local/share/julia:/home/ivan/.local/share/julia:/home/ivan/.local/share/julia:
```

**Comment by t-bltg** [comment]
Thanks for narrowing.

That's a though one since insets have been modified in `< 1.36.2` for `twin` axes in https://github.com/JuliaPlots/Plots.jl/pull/4488 it's quite hard to find a satisfactory fix...

And after that, padding changed in https://github.com/JuliaPlots/Plots.jl/pull/4535 to fix https://github.com/JuliaPlots/Plots.jl/issues/4533.

What is the output on `1.35.8` ?

**Comment by t-bltg** [comment]
The `lens` example seems broken since `1.36`: https://docs.juliaplots.org/v1.37/gallery/pgfplotsx/generated/pgfplotsx-ref040/#pgfplotsx_ref040, but other backends are unaffected.

**Comment by ivan-boikov** [comment]
> Thanks for narrowing.
> 
> That's a though one since insets have been modified in `< 1.36.2` for `twin` axes in #4488 it's quite hard to find a satisfactory fix...
> 
> And after that, padding changed in #4535 to fix #4533.
> 
> What is the output on `1.35.8` ?

It works, the output is identical to 1.36.2
![pic-selected-221207-2124-09](https://user-images.githubusercontent.com/66747290/206287938-d5b63f48-4681-4c1a-a08d-063cfa005968.png)

**Comment by t-bltg** [comment]
I guess https://github.com/JuliaPlots/Plots.jl/pull/4588 fixes it:
![jl_tmLM3QNipy](https://user-images.githubusercontent.com/13423344/206294013-20362949-911d-4603-8b2a-f6ae6838b679.png)
